internal/provider/metric_resource.go

```diff
@@ -0,0 +1,357 @@
+				Description: "The type of the metric.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(string(UptimeMetric), string(ResponseTimeMetric)),
+				},
+			},
+			"enabled": schema.BoolAttribute{
+				Description: "A flag indicating if the metric is enabled.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"visible": schema.BoolAttribute{
+				Description: "A flag indicating if the metric is visible.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"remote_id": schema.StringAttribute{
+				Description: "The remote ID for the metric.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"remote_name": schema.StringAttribute{
+				Description: "The remote name for the metric.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"threshold": schema.Int64Attribute{
+				Description: "The threshold value for the metric.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"featured_number": schema.StringAttribute{
+				Description: "A featured number for the metric.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(string(AvgFeatured), string(LastFeatured), string(MaxFeatured)),
+				},
+			},
+			"integration_id": schema.StringAttribute{
+				Description: "The integration ID related to the metric.",
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (r *MetricResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*statuspal.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf(
+				"Expected *statuspal.Client, got: %T. Please report this issue to the provider developers.",
+				req.ProviderData,
+			),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// https://www.statuspal.io/api-docs#tag/Metrics/operation/addMetric
+func (r *MetricResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data MetricResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Title.IsNull() || data.Title.IsUnknown() {
+		resp.Diagnostics.AddError(
+			"Missing Metric ID",
+			"Cannot create the metric because the ID field is missing from the plan.",
+		)
+
+		return
+	}
+
+	if data.Title.IsNull() || data.Title.IsUnknown() {
+		resp.Diagnostics.AddError(
+			"Missing Metric Title",
+			"Cannot create the metric because the Title field is missing or unknown in the plan.",
+		)
+
+		return
+	}
+
+	if data.Unit.IsNull() || data.Unit.IsUnknown() {
+		resp.Diagnostics.AddError(
+			"Missing Metric Unit",
+			"Cannot create the metric because the Unit field is missing or unknown in the plan.",
+		)
+
+		return
+	}
```

Comment:
Why we need these checks, since these attributes are set as required in the schema?